subsys/bluetooth/mesh/pb_adv.c

```diff
@@ -538,6 +538,8 @@ static void send_reliable(void)
 		} else {
 			bt_mesh_adv_send(buf, &buf_sent_cb, NULL);
 		}
+
+		net_buf_unref(buf);
```

Comment:
Shouldn't `link.tx.buf[i]` be set to NULL as well, since it seems that's the reference you're unrefing here?

Reply:
Agreed, I think this change is wrong, the buffers should be freed once the transaction has been acked in gen_prov_ack and link_ack, freeing them here removes the ability to retransmit on a nack.

Reply:
bt_mesh_adv_send() will call net_buf_ref every time a buffer is sent so if we retransmit and only unref (and then set it to NULL) in free_segments(), we are leaking them.

Reply:
But the ref in bt_mesh_adv_send will be unref'd in adv.c:adv_send, won't it? And the new unref on line 631 is to cancel out the adv_buf_create above it, or am I missing something here

Reply:
Yes, you are right. I overdid it. :smile: So only one unref was missing.